### PR TITLE
Fix #2220 Make composite disposible final inside the project

### LINKED ISSUE #2220: Make composite disposable final inside the whole project
**Actual Behaviour**

The CompositeDisposable's in the ScheduleFragment and MainActivity are not final. May be there are other instances of this too.

**Expected Behaviour**

They should be made final.

**Would you like to work on the issue?**

~~Yes~~ 
No
To be done by beginners (Less than 5 commits in this project)

### PR DESCRIPTION
**Pull request by vidu171**
Fixes #2220 

Changes: Ony two instances of composite disposable were present one inside Main Activity and other inside Schedule Fragment.

### CODE REVIEW

**Comment by dr0pdb** [comment]
Please resolve conflicts.

**Comment by iamareebjamal** [comment]
@vidu171 There's only one change in the PR. Also, remove null checks from final disposable if any

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 1]
```diff
@@ -865,7 +865,8 @@ public void run() {
     protected void onDestroy() {
         super.onDestroy();
         unbindService(customTabsServiceConnection);
-        disposable.dispose();
+        if( !disposable.isDisposed())
+            disposable.dispose();
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: No need for isDisposed check

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java (hunk 1 above)
Comment (vidu171) [reply android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: @iamareebjamal  noted thanks

**Comment by vidu171** [comment]
@iamareebjamal  done

**Comment by dr0pdb** [comment]
@vidu171 Please squash your commits.

**Update**
Also resolve conflicts.
That's why you shouldn't leave a PR pending. The more the other PRs are merged, the more updates you'll have to do.

**Comment by vidu171** [comment]
I have done the requested changes and squashed rhe commits too why am i still unable to merge?

**Comment by dr0pdb** [comment]
@vidu171 Please resolve the conflicts. After you opened the PR, a lot of changes have already been made to the repo so you're getting merge conflicts. You will need to resolve them before you can merge.

**Pull request closed without merging**